sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptionErrors.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Storage.Blobs.Specialized.Cryptography
+{
+    internal class EncryptionErrors
+    {
+        public static ArgumentException KeyNotFound(string keyId)
+            => new ArgumentException($"Could not resolve key of id `{keyId}`.");
+
+        public static ArgumentException BadEncryptionAgent(string agent)
+            => new ArgumentException("Invalid Encryption Agent. This version of the client library does not understand" +
+                $"the Encryption Agent `{agent}` set on the blob.");
+
+        public static ArgumentException BadEncryptionAlgorithm()
+            => new ArgumentException("Invalid Encryption Algorithm found on the resource. This version of the client" +
+                "library does not support the given encryption algorithm.");
+
+        public static ArgumentException NoKeyAccessor()
+            => new ArgumentException("Key and KeyResolver cannot both be null.");
```

Comment:
Nit - take these property names as parameters.  Pass them into the method using `nameof(...)` and then they'll be updated automatically when refactoring.

Reply:
Include the parameter names of everything if you're throwing ArgumentException